power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/base_optimizer.hpp

```diff
@@ -78,7 +79,8 @@ class NoOptimizer : public detail::BaseOptimizer<StateCalculator, State_> {
 
     NoOptimizer(Calculator func) : func_{std::move(func)} {}
 
-    auto optimize(State const& state, CalculationMethod method) -> MathOutput<ResultType> final {
+    auto optimize(State const& state, CalculationMethod method, bool /*extra_option*/ = true)
```

Comment:
Is this to be implemented later?

Reply:
`extra_option` is put in comment to avoid the "unused param" complain. It has to be present due to parent class. This is common practice.

Reply:
I guess I miss-worded my comment. My question really is, we intend to add `extra_option` at some point here (without the comment) to indicate that the binary search is to be used instead of the scanline method, correct? Hence, why I implied is not ready implemented.

Reply:
Not really. We will make a descision next week regarding this as it concerns interface. My proposal is to combine `CalculationMethod` with this one and make a new `OptimizationOption`. For now, it appears here because the `TapPositionOptimizer` will have to use an extra boolean, hence the parent class, hence here.

Reply:
I think your proposal should be the most straightforward way to go.

Reply:
i personally don't see the need for a boolean extra option. either you can make it an enum, or you can promote the `BinarySearch` to a generic search engine and pass it to the tap optimizer. (i remember you proposing that in one of the other comments on this PR but can't find it anymore) or is the logic in the tap optimizer too specific for that?

Reply:
See comment below

Reply:
addressed in recent commit.